include/eve/detail/kumi.hpp

```diff
@@ -485,6 +486,47 @@ namespace kumi
       (std::make_index_sequence<sizeof...(Ts)>());
     }
 
+    template<product_type Other>
+    friend constexpr auto operator<(tuple const &lhs, Other const &rhs) noexcept
+    {
+      // lexicographical order is defined as
+      // (v0 < w0) || ... andnot(wi < vi, vi+1 < wi+1) ... || andnot(wn-1 < vn-1, vn < wn);
+      auto res = get<0>(lhs) < get<0>(rhs);
+
+      auto const order = [&]<typename Index>(Index i)
+      {
+        auto y_less_x_prev  = rhs[i]  < lhs[i];
+        auto x_less_y       = lhs[index_t<Index::value+1>{}] < rhs[index_t<Index::value+1>{}];
+        res                 = res || (x_less_y && !y_less_x_prev);
+      };
+
+      [&]<std::size_t... I>(std::index_sequence<I...>)
+      {
+        (order(index_t<I>{}),...);
+      }
+      (std::make_index_sequence<sizeof...(Ts)-1>());
+
+      return res;
+    }
+
+    template<product_type Other>
+    friend constexpr auto operator<=(tuple const &lhs, Other const &rhs) noexcept
+    {
+      return (lhs < rhs) || (lhs == rhs);
```

Comment:
`!(rhs < lhs)`

Reply:
Or is it some Nan thing? I don't think that's a valid concern to sabotage perf for everyone.

Reply:
Nah, you do 'not' in other places, so it's not a Nan thing.
You can probably get an efficient Nan correctness if you just do a fold in `eve`

Reply:
`<=` is not equivalent to `! <`

The idea here was that, as KUMI needs the ordering, we can just piggy-back it.
I admit I have been slightly lazy here and should have written properly optimized >

Reply:
This is really not an OK implementation for eve. Should be done properly.

a <= b is 100% equivalent !(b < a). Unless you are talking Nan.

Reply:
ok I'm stupid, it's a lexical sort. Gonna fix it

Reply:
Fixed